control/mpc_follower/src/mpc_follower_core.cpp

```diff
@@ -803,78 +841,170 @@ void MPCFollower::callbackTrajectory(const autoware_planning_msgs::Trajectory::C
     mpc_traj_smoothed.yaw.back(), v_end, mpc_traj_smoothed.k.back(), t_end);
 
   if (!mpc_traj_smoothed.size()) {
-    ROS_INFO_COND(show_debug_info_, "[MPC] path callback: trajectory size is undesired.");
+    RCLCPP_INFO_EXPRESSION(
+      get_logger(), show_debug_info_, "path callback: trajectory size is undesired.");
     return;
   }
 
   ref_traj_ = mpc_traj_smoothed;
 
   /* publish debug marker */
-  visualization_msgs::MarkerArray markers;
-  std::string frame = msg->header.frame_id;
-  markers =
-    MPCUtils::convertTrajToMarker(mpc_traj_raw, "trajectory raw", 0.9, 0.5, 0.0, 0.05, frame);
-  pub_debug_marker_.publish(markers);
+  visualization_msgs::msg::MarkerArray markers;
+  const std::string & frame = msg->header.frame_id;
+  const builtin_interfaces::msg::Time & stamp = msg->header.stamp;
   markers = MPCUtils::convertTrajToMarker(
-    mpc_traj_resampled, "trajectory spline", 0.5, 0.1, 1.0, 0.05, frame);
-  pub_debug_marker_.publish(markers);
+    mpc_traj_raw, "trajectory raw", 0.9, 0.5, 0.0, 0.05, frame, stamp);
+  pub_debug_marker_->publish(markers);
   markers = MPCUtils::convertTrajToMarker(
-    mpc_traj_smoothed, "trajectory average filtered", 0.0, 1.0, 0.0, 0.05, frame);
-  pub_debug_marker_.publish(markers);
+    mpc_traj_resampled, "trajectory spline", 0.5, 0.1, 1.0, 0.05, frame, stamp);
+  pub_debug_marker_->publish(markers);
+  markers = MPCUtils::convertTrajToMarker(
+    mpc_traj_smoothed, "trajectory average filtered", 0.0, 1.0, 0.0, 0.05, frame, stamp);
+  pub_debug_marker_->publish(markers);
 }
 
 void MPCFollower::updateCurrentPose()
 {
-  geometry_msgs::TransformStamped transform;
+  geometry_msgs::msg::TransformStamped transform;
   try {
-    transform = tf_buffer_.lookupTransform("map", "base_link", ros::Time(0));
+    transform = tf_buffer_.lookupTransform("map", "base_link", rclcpp::Time(0));
   } catch (tf2::TransformException & ex) {
-    ROS_WARN_DELAYED_THROTTLE(
-      5.0, "[mpc_follower] cannot get map to base_link transform. %s", ex.what());
+    RCLCPP_WARN_SKIPFIRST_THROTTLE(
+      get_logger(), *get_clock(), 5.0, "cannot get map to base_link transform. %s", ex.what());
```

Comment:
I update the 5.0 to a duration in millisecond here and everywhere else

Reply:
Looks good! Thank you.